ppx/ppx_cstruct.ml

```diff
@@ -526,15 +526,20 @@ let constr_enum = function
   | {pcd_loc = loc; _} ->
     loc_err loc "invalid cenum variant"
 
+let get_len = function
+  | [ ({txt = "len"; _},
+       PStr
+         [{pstr_desc =
+             Pstr_eval ({pexp_desc = Pexp_constant (Pconst_integer (sz, None)); _}, _)
+          ; _}])] ->
+    Some (int_of_string sz)
+  | [{txt = "len"; loc}, _ ] ->
+    loc_err loc "[@len] argument should be an integer"
```

Comment:
```suggestion
    loc_err loc "[@len] argument should be a positive, non-zero integer"
```

Reply:
there are two check sites so I used this more specific version only when needed, how does that look?